protocols/luci-proto-wireguard/htdocs/luci-static/resources/protocol/wireguard.js

```diff
@@ -114,9 +139,24 @@ return network.registerProtocol('wireguard', {
 		o.validate = validateBase64;
 		o.rmempty = false;
 
-		o = ss.option(form.Value, 'preshared_key', _('Preshared Key'), _('Optional. Base64-encoded preshared key. Adds in an additional layer of symmetric-key cryptography for post-quantum resistance.'));
+		var preshared_key_files = {},
+		    comment = ' Adds in an additional layer of symmetric-key cryptography for post-quantum resistance.';
+		o = ss.option(form.FileUpload, 'preshared_key_file', _('Preshared Key file'), _('Optional either this or below. Preshared key file.' + comment));
```

Comment:
This will not work, the argument to `_(...)` *must* be a literal string - so use `comment = _('Adds an additional ...')` above and `_('Optional either this ...') + ' ' + comment` here.